solutions/python-igraph/clique_percolation/clique_percolation.py

```diff
@@ -0,0 +1,73 @@
+import igraph as ig
+import itertools
+import matplotlib.pyplot as plt
+
+EXAMPLE_K = 3
+
+def k_cliques(g, k):
+    """Returns a list of all cliques of size k in Graph g. Each clique is given as a set of vertex ids in g.
+    """
+    return g.cliques(k, k)
+
+def k_communities(g, k, min_common_vertices):
+    """
+    Given a graph and size k, return a list of communities found through clique
+    percolation with at least min_common_vertices shared.
+    A community is the connected component of cliques where two cliques overlap
+    in at least k-1 vertices.
+    Each community is a set of vertices in that community
+
+    Approach: generate a new clique graph where each k-clique is a vertex, and 
+    connect vertices if the cliques share min_common_vertices
+    """
+    if (min_common_vertices > k):
+        raise ValueError("Error: min_common_vertices=%d must be greater than k=%d" % (min_common_vertices, k))
+
+    # Each clique can be identified by it's index in 'cliques'
+    cliques = list(map(set, k_cliques(g, k)))
+    num_cliques = len(cliques)
+    # Graph storing each clique as a vertex
+    clique_graph = ig.Graph()
+    clique_graph.add_vertices(num_cliques)
+    # List of communities as a list of list of set of vertices
+    communities = []
+
+    # Get connected cliques
+    # If the two cliques share at least min_common_vertices,
+    # they are part of the same community
+    cg_edges = [(c1, c2) for c1, c2 in itertools.combinations(range(num_cliques), 2) if len(cliques[c1].intersection(cliques[c2])) >= min_common_vertices]
+
+    # Add edges for clique graph
+    clique_graph.add_edges(cg_edges)
```

Comment:
I think you would also be able to simply construct a graph directly here:
```suggestion
    clique_graph = ig.Graph(cg_edges)
```

That way, you don't first need to construct a graph, and call `add_vertices`.